src/qt/transactionrecord.cpp

```diff
@@ -40,19 +40,46 @@ bool TransactionRecord::showTransaction(const CWalletTx &wtx)
     return true;
 }
 
-	int64_t GetMyValueOut(const CWallet *wallet,const CWalletTx &wtx)
+int64_t GetMyValueOut(const CWallet *wallet, const CWalletTx &wtx)
+{
+    int64_t nValueOut = 0;
+    for (auto const& txout : wtx.vout)
     {
-        int64_t nValueOut = 0;
-        for (auto const& txout : wtx.vout)
+       if (wallet->IsMine(txout))
+       {
+            nValueOut += txout.nValue;
+       }
+    }
+    return nValueOut;
+}
+
+
+int64_t GetMyValueOut(const CWallet *wallet, const CWalletTx &wtx, unsigned int& index)
+{
+    int64_t nValueOut = 0;
+    if (wallet->IsMine(wtx.vout[index]))
+        nValueOut += wtx.vout[index].nValue;
+
+    return nValueOut;
+}
+
+unsigned int GetNumberOfStakeReturnOutputs(const CWallet *wallet,const CWalletTx &wtx)
+{
+    unsigned int nNumberOfOutputs = 0;
+
+    if (wtx.IsCoinStake())
+    {
+        for (auto const& txout: wtx.vout)
         {
-	       if (wallet->IsMine(txout))
-		   {
-				nValueOut += txout.nValue;
-		   }
+            // Count the number of outputs that have a pubkey equal to the first (non-empty) output.
+            // This are the stakesplits.
+            if (txout.scriptPubKey == wtx.vout[1].scriptPubKey)
+                nNumberOfOutputs++;
```

Comment:
This can fail, because on splitstake, the first output is P2PK, while the second is P2PKH. See https://hr.tbrada.eu/gbx/block/2ae492a0f1868d0b83b17b3b1073d04683f389be7bb8a683c9b0864489be2aa2

Reply:
I think that second output was a sidestake, not splitstake. The intention here is to merge split UTXO's NOT sidestake UTXOs. So I disagree with you.

For example, if there are three vouts that are UTXO splits (i.e. back to the same address as the input, or equivalently vout[1]), then we want those three merged together into a net transaction for display. Conversely, if there are sidestakes to other addresses those should be separated out. I have confirmed this to work properly in test,

Reply:
First part correct. The transaction was indeed side-stake. And I fully agree with the second paragraph.

What raised my concern was equivalence comparison of output scripts and the evidence that side-stake scripts are different than return-stake. And they would be unequal even if directed to the same address.

Reply:
Yes. And that would cause them to be printed out as separate "transactions" in the UI, which is exactly what we want. A sidestake looped back to the same address will be shown as a separate transaction from the (split) stake return...